pkg/kubelet/util/pluginwatcher/example_handler.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluginwatcher
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
+	v1beta1 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta1"
+	v1beta2 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta2"
+)
+
+type exampleHandler struct {
+	registeredPlugins        map[string]bool
```

Comment:
Hi @figo,

This field should be `registeredPlugins map[string]struct{}`. Always use an anonymous struct when creating maps that just indicate whether something exists or not. An anonymous struct takes up zero memory, whereas a `bool` does. For [example](https://play.golang.org/p/AzXEf7wJMd2):

```go
package main

import (
	"fmt"
	"unsafe"
)

func main() {
	b := true
	s := struct{}{}
	fmt.Printf("sizeof(bool)=%v\n", unsafe.Sizeof(b))
	fmt.Printf("sizeof(struct{})=%v\n", unsafe.Sizeof(s))
}
```

```shell
sizeof(bool)=1
sizeof(struct{})=0
```